api/v1beta1/gcpmachinetemplate_webhook.go

```diff
@@ -99,3 +101,17 @@ func (r *GCPMachineTemplate) ValidateDelete() error {
 func (r *GCPMachineTemplate) Default() {
 	clusterlog.Info("default", "name", r.Name)
 }
+
+func (r *GCPMachineTemplate) validateConfidentialCompute() error {
+	if r.Spec.Template.Spec.ConfidentialCompute != nil && *r.Spec.Template.Spec.ConfidentialCompute == ConfidentialComputePolicyEnabled {
```

Comment:
We can probably share this validation check with the GCPMachine. What do you think?